### ISSUE TITLE
First Option Issue

### ISSUE THREAD

**Issue by Sanderson10453** [reifjulian/strgroup#3]
Hi Prof. Reif,

Thanks for creating this great program! I'm currently using it for some matching within a large dataset and have run into an issue with the _first_ option. It seems that the first option isn't working and still groups observations even if their first character differs. I've been working around by making groups based upon the first character of the string, then running strgroup with the _by_ option.

I've attached an image to highlight the issue! I'm wondering if it's an issue with the program or maybe how I'm using the program.

Thanks for all your help

Best,
Spenser

![Strgroup_Github](https://user-images.githubusercontent.com/114104760/191597381-5bf7b1db-aef9-4bf8-81a9-44d46d0fc438.PNG)

**Comment by Sanderson10453** [comment]
> Hi Prof. Reif,
> 
> Thanks for creating this great program! I'm currently using it for some matching within a large dataset and have run into an issue with the _first_ option. It seems that the first option isn't working and still groups observations even if their first character differs. I've been working around by making groups based upon the first character of the string, then running strgroup with the _by_ option.
> 
> I've attached an image to highlight the issue! I'm wondering if it's an issue with the program or maybe how I'm using the program.
> 
> Thanks for all your help
> 
> Best, Spenser
> 
> ![Strgroup_Github](https://user-images.githubusercontent.com/114104760/191597381-5bf7b1db-aef9-4bf8-81a9-44d46d0fc438.PNG)

Also, I know this is different than the Statalist Q&A format - I can try to find a sysuse dataset that will highlight the issue if that's easiest

**Comment by reifjulian** [comment]
@Sanderson10453: Yes, please provide a minimum working example (eg with a built-in dataset) so that I can replicate the bug.
